src/main/java/com/icthh/xm/uaa/service/TenantPermissionService.java

```diff
@@ -39,16 +44,20 @@
     // Map structure: <Tenant, <Role, [Permission]>>
     final Map<String, Map<String, List<Permission>>> tenantRolePermissions = new ConcurrentHashMap<>();
 
-    public List<AccPermissionDTO> getEnabledPermissionByRole(String role) {
+    public List<AccPermissionDTO> getEnabledPermissionByRole(List<String> roles) {
 
         String tenant = TenantContextUtils.getRequiredTenantKeyValue(tenantContextHolder.getContext());
 
-        return tenantRolePermissions.getOrDefault(tenant, Collections.emptyMap())
-                                    .getOrDefault(role, Collections.emptyList())
-                                    .stream()
-                                    .filter(permission -> !permission.isDisabled())
-                                    .map(AccPermissionDTO::new)
-                                    .collect(Collectors.toList());
+        return ofNullable(roles).orElse(emptyList()).stream().map(role ->
+            tenantRolePermissions.getOrDefault(tenant, Collections.emptyMap())
+                .getOrDefault(role, emptyList())
+                .stream()
+                .filter(permission -> !permission.isDisabled())
+                .map(AccPermissionDTO::new)
+                .collect(Collectors.toList()))
+            .filter(CollectionUtils::isNotEmpty)
+            .findFirst()
```

Comment:
@ibilous Why did you decide to take just the first list of permission? If we have multi toles then I think we need to get all permission from all roles? @ishkurko @sergeysenja1992 @major13ua am I right?

Reply:
fixed